Build failure on Fedora 41 X86_64

Building `posix` is failing with the following linker error: I think you might need to add a compiler flag.

During the C phase of the build:

```
[ build ] make: Entering directory '/home/brandon/.pack/.tmp7/posix/posix/support'
[ build ] cc    -c -o posix.o posix.c
[ build ] cc -shared -fuse-ld=gold -o posix-idris posix.o
/usr/bin/ld.gold: error: posix.o: requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
collect2: error: ld returned 1 exit status
make: *** [Makefile:24: posix-idris] Error 1
```

During the Idris build phase:

```
[ build ] 38/38: Building System.Posix.Dir (src/System/Posix/Dir.idr)
/usr/bin/ld.gold: error: posix.o: requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
collect2: error: ld returned 1 exit status
make: *** [Makefile:24: posix-idris] Error 1
install: cannot stat 'posix-idris': No such file or directory
make: *** [Makefile:36: install] Error 1
Error: Script failed
```

--- a/posix/support/Makefile
+++ b/posix/support/Makefile
@@ -4,7 +4,7 @@ TARGET_VERSION ?= 0.0.1
 INSTALLDIR = ../lib
 
 LDFLAGS =
-CPPFLAGS =
+CPPFLAGS = -fPIC
 
 CC_VERSION = $(shell $(CC) --version)
 

